### PR TITLE
fix live chart bar width problem

### PR DESCRIPTION
**Pull request by ylwu-amzn**
*Issue #, if available:*

*Description of changes:*

* fix live chart bar width problem

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/utils/anomalyResultUtils.ts [hunk 1]
```diff
@@ -124,32 +124,42 @@ export const prepareDataForChart = (
...
+      confidence: null,
+      anomalyGrade: null,
+    });
+    anomalies.unshift({
+      startTime: dateRange.endDate,
+      endTime: dateRange.endDate,
+      plotTime: dateRange.endDate,
+      confidence: null,
+      anomalyGrade: null,
+    });
```
Comment (yizheliu-amazon) [inline public/pages/utils/anomalyResultUtils.ts:162]: will X Axis automatically shrink if only start/end data points are inserted? I recall I got such issue if plotting data only has start/end data.

Comment (yizheliu-amazon) [reply public/pages/utils/anomalyResultUtils.ts:162]: nvm. just realized you insert real data in above code

Comment (ylwu-amzn) [reply public/pages/utils/anomalyResultUtils.ts:162]: Good question, let me test locally.

File: public/pages/utils/anomalyResultUtils.ts [hunk 2]
```diff
@@ -124,32 +124,42 @@ export const prepareDataForChart = (
...
-    startTime: dateRange.endDate,
-    endTime: dateRange.endDate,
-    plotTime: interval
-      ? dateRange.endDate + MIN_IN_MILLI_SECS * interval
-      : dateRange.endDate,
-    confidence: null,
-    anomalyGrade: null,
-  });
-  if (getFloorPlotTime) {
+  if (getFloorPlotTime) { // we need to get floor plot time for bar chart
```
Comment (yizheliu-amazon) [inline public/pages/utils/anomalyResultUtils.ts:127]: question: In which cases `getFloorPlotTime ` is undefined?

File: public/pages/utils/anomalyResultUtils.ts (hunk 1 above)
Comment (yizheliu-amazon) [reply public/pages/utils/anomalyResultUtils.ts:162]: I think for your case it should be fine. Anomalies can only be empty when detector is not running, and in that case, we hide live chart.

File: public/pages/utils/anomalyResultUtils.ts (hunk 2 above)
Comment (yizheliu-amazon) [reply public/pages/utils/anomalyResultUtils.ts:127]: As discussed offline, there exists case where `getFloorPlotTime ` is not needed, aka LineSeries. I am good with that.

File: public/pages/utils/anomalyResultUtils.ts [hunk 3]
```diff
@@ -124,24 +124,53 @@ export const prepareDataForChart = (
...
+      anomalies.length > 0
+        ? anomalies[anomalies.length - 1].plotTime
+        : getFloorPlotTime(dateRange.startDate);
+    let endTime =
+      anomalies.length > 0
+        ? anomalies[0].plotTime
+        : getFloorPlotTime(dateRange.endDate);
+
+    while (endTime < dateRange.endDate) {
+      endTime += MIN_IN_MILLI_SECS * interval;
```
Comment (yizheliu-amazon) [inline public/pages/utils/anomalyResultUtils.ts:145]: add comment saying why we need this, such as: "make sure the end of plot time is big enough to cover NOW annotation, which is at dateRange.endDate"

Comment (ylwu-amzn) [reply public/pages/utils/anomalyResultUtils.ts:145]: sure

**Pull request merged**